storage/paths/index.go

```diff
@@ -575,6 +1290,81 @@ func (i *Index) StorageInfo(ctx context.Context, id storiface.ID) (storiface.Sto
 	return *si.info, nil
 }
 
+func (dbi *DBIndex) StorageBestAlloc(ctx context.Context, allocate storiface.SectorFileType, ssize abi.SectorSize, pathType storiface.PathType) ([]storiface.StorageInfo, error) {
+	var err error
+	var spaceReq uint64
+	switch pathType {
+	case storiface.PathSealing:
+		spaceReq, err = allocate.SealSpaceUse(ssize)
+	case storiface.PathStorage:
+		spaceReq, err = allocate.StoreSpaceUse(ssize)
+	default:
+		return nil, xerrors.Errorf("Unexpected path type")
+	}
+	if err != nil {
+		return nil, xerrors.Errorf("estimating required space: %w", err)
+	}
+
+	var checkString string
+	if pathType == storiface.PathSealing {
```

Comment:
This kinda stuff is supposed to be blocked by the db system. 
Consider instead:
     ptBool := pathType = storiface.PathSealing
     in SQL:
     WHERE ($1 AND can_seal=TRUE OR NOT $1 AND can_store=TRUE)
     (having $1 be ptBool)

Reply:
> This kinda stuff is supposed to be blocked by the db system.

Why is that?

The suggested query wouldn't have the same result as paths can have both CanSeal and CanStore set to true